pkg/server/handler.go

```diff
@@ -59,57 +59,66 @@ func (mc *MsgHandler) subscribeToMessages(msgChan chan network.InMsg) error {
 		err = mc.amqp.OnMessage(msgChan, queueName, exchange, key)
 	}
 
+	// Subscribe to messages received from any client
 	subscribe(msgChan, queueNameFogIn, exchangeFogIn, bindingKeyDevice)
 	subscribe(msgChan, queueNameFogIn, exchangeFogIn, bindingKeySchema)
 	subscribe(msgChan, queueNameFogIn, exchangeFogIn, bindingKeyDeviceCommands)
+
+	// Subscribe to messages received from the connector service
 	subscribe(msgChan, queueNameConnOut, exchangeConnOut, bindingKeyData)
 	subscribe(msgChan, queueNameConnOut, exchangeConnOut, bindingKeyDevice)
+
 	return err
 }
 
 func (mc *MsgHandler) onMsgReceived(msgChan chan network.InMsg) {
 	for {
+		var err error
 		msg := <-msgChan
 		mc.logger.Infof("Exchange: %s, routing key: %s", msg.Exchange, msg.RoutingKey)
 		mc.logger.Infof("Message received: %s", string(msg.Body))
 
-		authorizationHeader := msg.Headers["Authorization"]
-
-		switch msg.RoutingKey {
-		case "device.register":
-			err := mc.thingController.Register(msg.Body, authorizationHeader.(string))
-			if err != nil {
-				mc.logger.Error(err)
-				continue
-			}
-		case "device.registered":
-			// Ignore message
+		if msg.Exchange == exchangeFogIn {
+			err = mc.handleClientMessages(msg)
+		} else if msg.Exchange == exchangeConnOut {
+			err = mc.handleConnectorMessages(msg)
+		}
+
+		if err != nil {
+			mc.logger.Error(err)
 			continue
-		case "device.cmd.list":
-			mc.logger.Info("List things request received")
-			err := mc.thingController.ListDevices(authorizationHeader.(string))
-			if err != nil {
-				mc.logger.Error(err)
-				continue
-			}
-		case "device.cmd.auth":
-			err := mc.thingController.AuthDevice(msg.Body, authorizationHeader.(string))
-			if err != nil {
-				mc.logger.Error(err)
-				continue
-			}
-		case "data.request":
-			err := mc.thingController.RequestData(msg.Body, authorizationHeader.(string))
-			if err != nil {
-				mc.logger.Error(err)
-				continue
-			}
-		case "schema.update":
-			err := mc.thingController.UpdateSchema(msg.Body, authorizationHeader.(string))
-			if err != nil {
-				mc.logger.Error(err)
-				continue
-			}
 		}
 	}
 }
+
+func (mc *MsgHandler) handleClientMessages(msg network.InMsg) error {
+	authorizationHeader := msg.Headers["Authorization"]
+
+	switch msg.RoutingKey {
+	case "device.register":
+		return mc.thingController.Register(msg.Body, authorizationHeader.(string))
+	case "schema.update":
+		return mc.thingController.UpdateSchema(msg.Body, authorizationHeader.(string))
+	case "device.cmd.auth":
+		return mc.thingController.AuthDevice(msg.Body, authorizationHeader.(string))
+	case "device.cmd.list":
+		return mc.thingController.ListDevices(authorizationHeader.(string))
+	}
+
+	return nil
+}
+
+func (mc *MsgHandler) handleConnectorMessages(msg network.InMsg) error {
+	authorizationHeader := msg.Headers["Authorization"]
+
+	switch msg.RoutingKey {
+	case "data.request":
+		return mc.thingController.RequestData(msg.Body, authorizationHeader.(string))
+	case "device.cmd.list":
```

Comment:
Should this command been treated by connector message handler function too?? This documentation [here](https://github.com/CESARBR/knot-fog-connector/blob/master/docs/api/amqp.md#amqp-api) indicate that the `device.cmd.list` message would be sent only by client side.

If it should be here either, there is a missing subscribe around line 70:
`subscribe(msgChan, queueNameConnOut, exchangeConnOut, bindingKeyDeviceCommands)`

Reply:
Good catch, @esdrasjnr. Actually, it was a mistake. At least for now, we don't need to handle that request, it will be included only if the connector requires it to do some diff between the devices registered on the fog or cloud.